test/chromium-test.js

```diff
@@ -446,37 +446,36 @@ describe('chromium', function () {
 
   context('https-server with a port value given', function () {
     var server;
+    var port;
 
     before(function (done) {
       server = net.createServer();
-      server.listen(3001, function (err) {
+      server.listen(function (err) {
+        if (!err) {
+          port = String(server.address().port);
+        }
         done(err);
       });
     });
 
-    after(function () {
-      server.close();
+    after(function (done) {
+      server.close(done);
```

Comment:
Does this mean the flaky tests were a race condition between a server starting and another one potentially not being closed yet?

Reply:
Yes. At least after upgrading puppeteer that was the only randomly failing issue. We had other issues before as well.